dotnet/src/SemanticKernel.Abstractions/Services/OrderedAIServiceSelector.cs

```diff
@@ -20,7 +21,10 @@ internal sealed class OrderedAIServiceSelector : IAIServiceSelector
         [NotNullWhen(true)] out T? service,
         out PromptExecutionSettings? serviceSettings) where T : class, IAIService
     {
-        var executionSettings = function.ExecutionSettings;
+        // Allow the execution settings from the kernel arguments to take precedence
+        var executionSettings = arguments.ExecutionSettings is not null
+             ? new List<PromptExecutionSettings> { arguments.ExecutionSettings }
```

Comment:
I had to do this

```csharp
List<PromptExecutionSettings>? executionSettings = arguments.ExecutionSettings is null
     ? function.ExecutionSettings
     : new List<PromptExecutionSettings>(new PromptExecutionSettings[] { arguments.ExecutionSettings });
```

To avoid CS0029 Cannot implicitly convert type 'type' to 'type'

So will leave as is for now